packages/headless/src/features/case-assist/case-assist-analytics-actions-loader.ts

```diff
@@ -0,0 +1,98 @@
+import {AsyncThunkAction} from '@reduxjs/toolkit';
+import {StateNeededByCaseAssistAnalytics} from '../../api/analytics/analytics';
+import {CaseAssistEngine} from '../../app/case-assist-engine/case-assist-engine';
+import {AsyncThunkAnalyticsOptions} from '../analytics/analytics-utils';
+import {
+  logAbandonCase,
+  logCaseNextStage,
+  logCaseStart,
+  logClassificationClick,
+  logCreateCase,
+  logDocumentSuggestionClick,
+  logDocumentSuggestionRating,
+  logSolveCase,
+  logUpdateCaseField,
+} from './case-assist-analytics-actions';
+
+export interface CaseAssistAnalyticsActionCreators {
+  logCaseStart(): AsyncThunkAction<
+    {},
+    void,
+    AsyncThunkAnalyticsOptions<StateNeededByCaseAssistAnalytics>
+  >;
+
+  logCaseNextStage(): AsyncThunkAction<
+    {},
+    void,
+    AsyncThunkAnalyticsOptions<StateNeededByCaseAssistAnalytics>
+  >;
+
+  logCreateCase(): AsyncThunkAction<
+    {},
+    void,
+    AsyncThunkAnalyticsOptions<StateNeededByCaseAssistAnalytics>
+  >;
+
+  logSolveCase(): AsyncThunkAction<
+    {},
+    void,
+    AsyncThunkAnalyticsOptions<StateNeededByCaseAssistAnalytics>
+  >;
+
+  logAbandonCase(): AsyncThunkAction<
+    {},
+    void,
+    AsyncThunkAnalyticsOptions<StateNeededByCaseAssistAnalytics>
+  >;
+
+  logUpdateCaseField(
+    fieldName: string
+  ): AsyncThunkAction<
+    {},
+    void,
+    AsyncThunkAnalyticsOptions<StateNeededByCaseAssistAnalytics>
+  >;
+
+  logClassificationClick(
+    classificationId: string
+  ): AsyncThunkAction<
+    {},
+    void,
+    AsyncThunkAnalyticsOptions<StateNeededByCaseAssistAnalytics>
+  >;
+
+  logDocumentSuggestionClick(
+    suggestionId: string
+  ): AsyncThunkAction<
+    {},
+    void,
+    AsyncThunkAnalyticsOptions<StateNeededByCaseAssistAnalytics>
+  >;
+
+  logDocumentSuggestionRating(
+    suggestionId: string,
+    rating: number
+  ): AsyncThunkAction<
+    {},
+    void,
+    AsyncThunkAnalyticsOptions<StateNeededByCaseAssistAnalytics>
+  >;
+}
+
+export function loadCaseAssistAnalyticsActions(
+  engine: CaseAssistEngine
+): CaseAssistAnalyticsActionCreators {
+  engine.addReducers({});
```

Comment:
If no reducers are needed, perhaps this line could be removed?

Reply:
I took example on other analytics actions loader. There all doing the same call. I guess it is to suppress this warning:

    'engine' is defined but never used.eslint@typescript-eslint/no-unused-vars

Reply:
Renaming the parameter to `_engine` works too.

Reply:
Oh, we can leave it in that case. I felt there might be something to it. As you prefer,